veps/sig-compute/image-volume-proposal.md

```diff
@@ -0,0 +1,169 @@
+- Make sure virt-launcher containers don’t get extra privileges, keeping the system secure.
+
+## Non Goals
+
+- This enhancement will not introduce changes to API fields or the API in general.
+- Users will not need to be aware of the underlying implementation details of container disks or kernel boot.
+- No additional functionality will be added to the existing API.
+- The current behavior, from users perspective, will remain unchanged.
+
+## Definition of Users
+
+- Anyone who creates virtual machines with container disk or kernel boot.
+- KubeVirt developers who need to develop features and fix bugs based on the current workaround.
+
+## User Stories
+
+### Faster Start-Up
+As a KubeVirt user, I want faster virtual machine start-ups, so that I can save time and improve system performance.
+
+### Easier Development for New Contributors
+As a new KubeVirt contributor, I want a simpler codebase, so that I can quickly understand how things work and contribute 
+to fixing bugs and adding features.
+
+### Better Security
+As a KubeVirt user, I want to remove privileged operations like bind mounts from virt-handler.
+
+## Repos
+
+https://github.com/kubevirt/kubevirt
+
+## Design
+
+This design introduces a new method for handling container disks and kernel boot. When a Virtual Machine with a container 
+disk or kernel boot is created, the virt-launcher pod will include an imageVolume in its specification, replacing the 
+current method that uses additional init containers and sidecar containers.
+
+Virt-handler will no longer be part of this process, and there will be no need to manage bind mounts or search for sockets. 
+From the launcher pod’s perspective, it will access the disk or kernel boot files directly from a predefined image path in 
+the volume mounts of virt-launcher.
+
+The design will ensure that the current method for container disks and kernel boot continues to work by default. 
+The new method using imageVolumes will only be used once the feature gate is enabled. By default, the feature 
+gate will be off until it is in Beta stage.
+
+Once the feature gate is graduated, we will ensure the old code is removed from virt-controller and virt-launcher. 
+For virt-handler, we will remove the code for initiating new VMs with container disks but will keep the code needed to 
+support existing VMs. Once we are confident that all virtual machines are using the new method and there are no upgrade 
+risks, we will completely remove the old code for supporting VMs from virt-handler.
+
+## API Examples
+
+As stated in the No Goals section, no API changes are included.
+
+## Alternatives
+
+One alternative to remove the current workaround was explored in [this PR](https://github.com/kubevirt/kubevirt/pull/11845). 
+It suggested using shared namespaces between the launcher containers to avoid bind mounts. However, 
+this approach may compromise the isolation of the compute container where the domain runs, as discussed 
```

Comment:
I don't believe that this would have been an issue. But OCI volumes is basically what we have hoped for, for years. If people can live with our less nice bind-mount solution in LTS versions which don't have OCI volumes yet in beta, I don't see a reason to implement it.

This proposal definitely defines the state where we wanted to be since we started kubevirt, so no need for any justification in that context. Great to see this!

Reply:
+1